code/modules/food_and_drinks/kitchen_machinery/grill.dm

```diff
@@ -0,0 +1,110 @@
+//I JUST WANNA GRILL FOR GOD'S SAKE
+
+/obj/machinery/grill
+	name = "grill"
+	desc = "Just like the old days."
+	icon = 'icons/obj/kitchen.dmi'
+	icon_state = "grill_open"
+	density = TRUE
+	layer = BELOW_OBJ_LAYER
+	use_power = NO_POWER_USE
+	var/grill_fuel = 0
+	var/grilling
+	var/datum/looping_sound/grill/grill_loop
+
+/obj/machinery/grill/Initialize()
+	..()
+	grill_loop = new(list(src), FALSE)
+
+/obj/machinery/grill/update_icon()
+	if(grilling)
+		icon_state = "grill"
+	else if(!grill_fuel)
+		icon_state = "grill_open"
+	else
+		icon_state = "grill_on"
+
+/obj/machinery/grill/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/trash/coal))
+		qdel(I)
+		to_chat(user, "<span class='notice'>You put the [I] in [src].</span>")
+		grill_fuel += 500
+		update_icon()
+		return
+	if(istype(I, /obj/item/stack/sheet/mineral/wood))
+		qdel(I)
+		to_chat(user, "<span class='notice'>You put the [I] in [src].</span>")
+		var/obj/item/stack/S = I
+		grill_fuel += (50 * S.amount)
+		update_icon()
+		return
+	if(I.resistance_flags & INDESTRUCTIBLE)
+		to_chat(user, "<span class='warning'>You don't feel it would be wise to grill [I]...</span>")
+		return ..()
+	if(istype(I, /obj/item/reagent_containers))
+		if(istype(I, /obj/item/reagent_containers/food) && !istype(I, /obj/item/reagent_containers/food/drinks))
+			if(HAS_TRAIT(I, TRAIT_NODROP) || (I.item_flags & (ABSTRACT | DROPDEL)))
+				return ..()
+			else if(!grill_fuel)
+				to_chat(user, "<span class='notice'>There is not enough fuel.</span>")
+				return
+			else if(!grilling && user.transferItemToLoc(I, src))
+				grilling = I
+				to_chat(user, "<span class='notice'>You put the [I] on [src].</span>")
+				update_icon()
+				grill_loop.start()
+				return
+		else
```

Comment:
check if it actually has the reagent first.

Reply:
why? doesn't runtime or anything and seems to work properly

Reply:
looks nicer imo

Reply:
bruh